### ISSUE TITLE
Train 2 gaminet models fails

### ISSUE THREAD

**Issue by JDE65** [interpretml/interpret#427]
I try to run in the same snippet two similar GamiNet models on the same dataset.
The first attempt works well but when I want to train the second model, it fails with a message: "element 0 of tensors does not require grad and does not have grad_fn".
The issues appears to be linked to the torch.autograd.backward()  called in 8tensor.py() on line 488, itself called by loss.backward of gaminet/base.py

i tried to invert the order, invert the dataset. With any combinaison, I get the same error with the second model.
I also tried to train twice the same model on the same data, and the same error appears.
It looks like a GamiNet model cannot be trained twice (or more) in a same script.
Could you please help?
Thanks
